### ISSUE TITLE
Laravel 4: filter method and column search

### ISSUE THREAD

**Issue by mifas** [yajra/laravel-datatables#134]
Hi, I'm using Laravel 4 branch, and using `filter` methods to do column search. following is my code

```
->filter(function($query){
                foreach( \Input::get('columns') as $column ){
                    if( strlen($column['search']['value']) > 0 ) {
                        $keyword = trim($column['search']['value']);

                        switch( $column['name'] ){
                            case 'birthday':
                                $crv = ( preg_match('/^[yes]+/i', $keyword) ) ? 1 : (( preg_match('/^[no]+/i', $keyword) ) ? 0 : 2);
                                $query->where('birthday', 'LIKE', "%{$crv}%");
                                break;
                            case 'expiry_date':
                                $redv = ( preg_match('/^[yes]+/i', $keyword) ) ? 1 : (( preg_match('/^[no]+/i', $keyword) ) ? 0 : 2);
                                $query->where('expiry_date', '=', "$redv");
                                break;
                            case 'start_date':
                                $query = dataTableDateSearch('start_date',$keyword,$query);
                                break;
                            case 'end_date':
                                $query = dataTableDateSearch('end_date',$keyword,$query);
                                break;
                            default:
                                $query->where($column['name'], 'LIKE', "%{$keyword}%");
                        }
                    }
                }
            })
```

problem is, the global search(main search) is not working. as far I know, `filter` method is overwriting everything, i want to my both search should work. Is there any options?

I'm using Laravel 4

**Comment by yajra** [comment]
@mifas, unfortunately you cannot use `filter` and global search at the same time. You would have to write all your query inside the filter closure.

**Comment by mifas** [comment]
Is there any way to override `filter` method?

**Comment by yajra** [comment]
Laravel 4 version has limited features. `filter` will override global search and that's it. You would have to write your custom queries inside its closure.

**Comment by mifas** [comment]
If would be nice, option like `->enableColumnSearch(true)`.?

**Comment by yajra** [comment]
@mifas, `->enableColumnSearch(true)` seems like a good idea or maybe something like `->filter(callback, true)` where 2nd param will enable column search. Will try to work on it on my free time or you can submit a PR if you can. Thanks!

**Comment by mifas** [comment]
@yajra  Thank you :) :+1:

**Comment by yajra** [comment]
@mifas, sorry for the delay on this one. As I am reviewing this enhancement, it appears that I already fixed this issue on the latest version of package. Anyways, columnSearch should always be called even if global search was overridden. Released on v3.6.9 for L4 version. Thanks!

**Comment by dtbarne** [comment]
This appears to be an issue as of v5.12.1. Global search won't work if I'm also using a filter().

**Comment by yajra** [comment]
@dtbarne YES, global search will not work when you use `filter` method. It was by design to disable global search and make your own custom query using filter closure.

**Comment by paulcanning** [comment]
@yajra it looks like you did a fix to allow the global search and custom filter method in auguast 2015, but then in November, said it won't work still. Which is it?

> Anyways, columnSearch should always be called even if global search was overridden. Released on v3.6.9 for L4 version

For the Laravel v5 library, it seems to not be the case.
